tools/editors-libraries.adoc

```diff
@@ -0,0 +1,177 @@
+<6> `onError` (optional, defaults to `() => {}`): If there's an error opening the Editor, this function will be called.
+<7> `resources` (optional, defaults to an empty Map): Map of resources that will be provided for the Editor. This can be used, for instance, to provide included models for the DMN Editor. Each entry in the map has the resource POSIX path (similar to the `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot` parameter) as its key and an object containing the `content-type` (`text` or `binary`) and the resource `content` (Promise similar to the `initialContent` parameter) as its value. Resources located in a parent directory (in relation to the current content path, defined by the `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot`) won't be listed to be used as an Included Model. Examples:
+ * `resources: new Map([["types.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], [...], [...]])`
+ * `resources: new Map([["/types/typeA.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], ["/extras/decision.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], [...]])`
+
+NOTE: The `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot` parameter is only available for the new DMN Editor. The classic editors don't have the concept of relative paths, so every mapped resource is available.
+
+The returned object (`newDmnEditor`) will contain the methods needed to manipulate the Editor:
+
+- `getContent(): Promise<string>`: Returns a Promise containing the Editor's content.
+- `setContent(normalizedPosixPathRelativeToTheWorkspaceRoot: string, content: string): Promise<void>`: Sets the content of the editor. The returning Promise will be rejected if setting the content fails.
+- `getPreview(): Promise<string>`: Returns a Promise containing the SVG string of the current diagram.
+- `subscribeToContentChanges(callback: (isDirty: boolean) => void): (isDirty: boolean) => void`: Setup a callback to be called on every content change in the Editor. Returns the same callback to be used for unsubscription.
+- `unsubscribeToContentChanges(callback: (isDirty: boolean) => void): void`: Unsubscribes the passed callback from content changes.
+- `markAsSaved(): void`: This resets the editor state, signaling that its content is saved. It also fires the subscribed callbacks for content changes.
+- `undo(): void`: Undo the last change in the Editor. It will also fire the subscribed callbacks of content changes.
+- `redo(): void`: Redo the last undone change in the Editor. It will also fire the subscribed callbacks of content changes.
+- `close(): void`: Closes the Editor.
+- `envelopeApi: MessageBusClientApi<KogitoEditorEnvelopeApi>`: Advanced Editor API.
+
+NOTE: For now, the new DMN Editor doesn't expose the `canvas` property.
```

Comment:
For now, the new DMN Editor doesn't expose the `canvas` property -> For now, the new DMN editor does not expose the 'canvas' property.